packages/account/src/providers/provider.ts

```diff
@@ -679,11 +705,12 @@ export default class Provider {
    * @hidden
    */
   private createOperations(): SdkOperations {
-    const fetchFn = Provider.getFetchFn(this.options);
+    const fetchFn = this.getFetchFn(this.options);
     const gqlClient = new GraphQLClient(this.urlWithoutAuth, {
       fetch: (input: RequestInfo | URL, requestInit?: RequestInit) =>
         fetchFn(input.toString(), requestInit || {}, this.options),
       responseMiddleware: (response: GraphQLClientResponse<unknown> | Error) => {
+        this.setCurrentBlockHeight(response as { extensions: unknown });
```

Comment:
@Torres-ssf Because the node waiting time threshold is dynamic and can be changed or disabled at any time, we must validate the response and implement the whole timing/check/retry independently.